### PR TITLE
Add support for CMake Presets V5

### PR DESCRIPTION
**Pull request by nasinlock**
## This change addresses item #2979

The following changes are proposed:

- Add support for the 'test output truncation' feature
- Add support for the 'path list separator macro.

## Other Notes/Information

Updates to the tests to account for these new features will be added in a second pull request.

### CODE REVIEW

File: test/extension-tests/successful-build/test/variable-substitution.test.ts [hunk 1]
```diff
@@ -138,6 +139,12 @@ suite('Variable Substitution', () => {
                 '[cmakeInstallPrefix] substitution incorrect');
         expect(typeof cacheEntry.value).to.eq('string', '[cmakeInstallPrefix] unexpected cache entry value type');
 
+        // Check substitution for "pathListSep".
+        cacheEntry = cache.get('pathListSep') as CacheEntry;
```
Comment (bobbrow) [inline test/extension-tests/successful-build/test/variable-substitution.test.ts:143]: I think you should revert the changes to this test file.  pathListSep only exists in the context of presets, it's not something you can use for substitution in the settings.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ## 1.14
 Improvements:
 - Add an optional description field to kits. [PR #2944](https://github.com/microsoft/vscode-cmake-tools/pull/2944) [@TisziV](https://github.com/TisziV)
+- Add support for CMake Presets V5 [#2979](https://github.com/microsoft/vscode-cmake-tools/issues/2979)
```
Comment (bobbrow) [inline CHANGELOG.md:6]: nit: please add a period at the end of the sentence.

**Review by bobbrow — CHANGES_REQUESTED** [review]

File: src/cmakeProject.ts [hunk 3]
```diff
@@ -1177,7 +1177,7 @@ export class CMakeProject {
             // recursively search the build directory for all
             const searchRoot = await this.binaryDir;
             if (await fs.exists(searchRoot)) {
-                (await fs.walk(searchRoot)).forEach(e => {
+                (await fs.walk(searchRoot)).forEach((e: any) => {
```
[the remark this answers is not in the document]
Comment (nasinlock) [reply src/cmakeProject.ts:1180]: I was getting an error reported for that line. I'll remove the change and see if it re-appears.

File: src/preset.ts [hunk 4]
```diff
@@ -590,6 +591,9 @@ async function getExpansionOptions(workspaceFolder: string, sourceDir: string, p
     if (preset.__file && preset.__file.version >= 4) {
         expansionOpts.vars['fileDir'] = path.dirname(preset.__file!.__path!);
     }
+    if (preset.__file && preset.__file.version >= 5) {
+        expansionOpts.vars['pathListSep'] = '${pathListSep}';
```
[the remark this answers is not in the document]
Comment (nasinlock) [reply src/preset.ts:595]: That makes sense. I thought in VSCode we let CMake handle the macro expansion. I'll update to do the expansion ourselves

[the remark this answers is not in the document]
Comment (nasinlock) [reply src/preset.ts]: Updated

File: test/extension-tests/successful-build/test/variable-substitution.test.ts [hunk 5]
```diff
@@ -47,7 +47,8 @@ suite('Variable Substitution', () => {
                 test1: true,
                 test2: 123,
                 test3: ["1", "2", "3"],
-                test4: {"type": "PATH", "value": "/usr/bin"}
+                test4: {"type": "PATH", "value": "/usr/bin"},
+                pathListSep: '${pathListSep}'
```
Comment (bobbrow) [inline test/extension-tests/successful-build/test/variable-substitution.test.ts:51]: revert this case too

**Review by elahehrashedi — APPROVED** [review]

**Review by bobbrow — APPROVED** [review]

**Pull request merged**